vidarr-prometheus/src/main/java/ca/on/oicr/gsi/vidarr/prometheus/PrometheusAlertManagerConsumableResource.java

```diff
@@ -0,0 +1,167 @@
+  }
+
+  private Set<String> fixedVidarrNames;
+  private String alertmanagerUrl;
+  private String environment;
+  private Set<String> labels;
+  @JsonIgnore private AlertCache cache;
+
+  public PrometheusAlertManagerConsumableResource() {}
+
+  public Set<String> getFixedVidarrNames() {
+    return fixedVidarrNames;
+  }
+
+  public String getAlertmanagerUrl() {
+    return alertmanagerUrl;
+  }
+
+  public String getEnvironment() {
+    return environment;
+  }
+
+  public Set<String> getLabels() {
+    return labels;
+  }
+
+  @Override
+  public void startup(String name) {
+    if (fixedVidarrNames.isEmpty()) {
+      throw new IllegalArgumentException(
+          "Fixed Vidarr names is empty in Prometheus Alertmanager resource config.");
+    }
+    if (alertmanagerUrl == null || "".equals(alertmanagerUrl)) {
+      throw new IllegalArgumentException(
+          "Alertmanager URL is missing in Prometheus Alertmanager resource config.");
+    }
+    if (environment == null || "".equals(environment)) {
+      throw new IllegalArgumentException(
+          "Environment is missing in Prometheus Alertmanager resource config.");
+    }
+    if (labels.isEmpty()) {
+      throw new IllegalArgumentException(
+          "Labels are missing in Prometheus Alertmanager resource config.");
+    }
+    cache = new AlertCache(name);
+  }
+
+  @Override
+  public Optional<Pair<String, BasicType>> inputFromSubmitter() {
+    return Optional.of(new Pair<>("required_services", BasicType.JSON));
+  }
+
+  @Override
+  public void recover(String workflowName, String workflowVersion, String vidarrId) {
+    // Do nothing, as there's no intermediate state that needs to be tracked
+  }
+
+  @Override
+  public void release(String workflowName, String workflowVersion, String vidarrId) {
+    // Do nothing, as this doesn't actually hold onto any resources
+  }
+
+  private Set<String> attemptDeserialization(JsonNode nodeInput) {
+    try {
+      return MAPPER.readValue(nodeInput.toString(), new TypeReference<>() {}); // Jackson dark magic
+    } catch (JsonProcessingException e) {
+      e.printStackTrace();
+
+      throw new IllegalArgumentException(e);
+    }
+  }
+
+  @Override
+  public ConsumableResourceResponse request(
+      String workflowName, String workflowVersion, String vidarrId, Optional<JsonNode> input) {
+    Set<String> inputRequiredServices;
+    try {
+      inputRequiredServices =
+          input.map(this::attemptDeserialization).orElse(Collections.emptySet());
+    } catch (IllegalArgumentException e) {
+      return ConsumableResourceResponse.error(
+          "Error reading required_services input from "
+              + "submitter. This should be a JSON list of strings.");
+    }
+
+    final var isInhibited =
+        cache
+            .get()
+            .flatMap(
+                alert ->
+                    alert.matches(
+                        environment,
+                        Stream.concat(
```

Comment:
This is fine, but since `concat` only takes two arguments, there is another way:

```
Stream.of(
  inputRequiredServices.stream(),
  fixedVidarrNames.stream(),
  Stream.of(workflowName))
.flatMap(Function.identity())
```

This creates a stream of streams and then flattens it. It's incredibly obtuse, but, here we are.

Reply:
And yet, less obtuse than the double `concat`. Updated!